### ISSUE TITLE
Consider allowing an array of colors for palette shifting

### ISSUE THREAD

**Issue by rparrett** [Lommix/bevy_nine_slice_ui#6]
I'm just now getting around to checking out the 0.5 update. Thanks for getting that out mid-jam, but I was too busy to even think about it.

Maybe this is only really useful for the very simple two-color graphics I was using and I haven't fully thought this through, but the main draw of the potential lookup gradient feature for me was dealing with fewer image assets. So the current implementation didn't seem worth using in my case.

Now that I'm thinking about this a little more though, I guess it would be super easy to build the palette assets in code, but I guess that's not the most intuitive API.

If you've already considered this, then I consider the issue resolved, so feel free to close.

**Comment by Lommix** [comment]
Totally agree, having the palette as `Vec<Color>`would be nice and very intuitive. But it is not that easy to do in code.
Each shader parameter has to be bound to a space in memory of a fixed size for the GPU to use with the exception being images, which can be bound dynamically.
And since I do not want to set a limit on the colors someone else might use, the best way was to just use an image.

I am by no means a WGPU expert, maybe there is a way I'm unaware of. But one thing I am sure about, this would not work with the given UI Material Pipeline from Bevy and would at least require a custom render pipeline.

Bevy Images do not require an actual image file. They are just a slice of data and can be created whenever you want it in code.

Here is an example how to generate a palette one on the fly:
```rust
    let colors = [
        Color::RED,
        Color::GREEN,
        Color::BLUE,
        Color::YELLOW,
    ]
    .iter()
    .flat_map(|c| c.as_rgba_u8())
    .collect::<Vec<_>>();

    let image = Image::new_fill(
        Extent3d {
            width: 4,
            height: 1,
            ..default()
        },
        TextureDimension::D2,
        colors.as_slice(),
        TextureFormat::Rgba8Unorm,
    );

    // mut images: ResMut<Assets<Image>>,
    let image_handle = images.add(image);
...
```
I leave the topic open. Maybe someone smarter has a better idea at some point.

**Comment by rparrett** [comment]
Yep, that's the exact idea I had mid-writing-this-issue, and I had assumed there was a technical reason it was done this way. Cool. Thanks for writing out that example.
